storm-core/src/clj/org/apache/storm/util.clj

```diff
@@ -553,7 +553,8 @@
                (catch InterruptedException e
                  (log-message log-prefix " interrupted.")))
              (exit-code-callback (.exitValue process)))
-           nil)))                    
+           nil)
+          :kill-fn (fn [error] (log-error (RuntimeException. ("launch process failed..."))))))
```

Comment:
Why are we throwing away error?

What happens if this does not exit?  I assume that the heartbeat monitoring will detect it and do the right thing, but are there other places in the code besides a worker process that we use this?  Will there be problems especially around the exit-code-callback if they are not called?